src/ui/menu_nbgl.c

```diff
@@ -25,46 +25,151 @@
                      app_quit);
 }
 
-// 'About' menu
+//  ----------------------------------------------------------- 
+//  --------------------- SETTINGS MENU -----------------------
+//  ----------------------------------------------------------- 
 
 static const char* const INFO_TYPES[] = {"Version", "Developer"};
 static const char* const INFO_CONTENTS[] = {APPVERSION, "Ledger"};
 
+// settings switches definitions 
+enum 
+{
+    DUMMY_SWITCH_1_TOKEN = FIRST_USER_TOKEN,
+    DUMMY_SWITCH_2_TOKEN
+};
+
+enum 
+{
+    DUMMY_SWITCH_1_ID = 0,
+    DUMMY_SWITCH_2_ID, 
+    SETTINGS_SWITCHES_NB
+};
+
+static nbgl_layoutSwitch_t switches[SETTINGS_SWITCHES_NB] = {0};
+
 static bool nav_callback(uint8_t page, nbgl_pageContent_t* content) {
     UNUSED(page);
-    content->type = INFOS_LIST;
-    content->infosList.nbInfos = 2;
-    content->infosList.infoTypes = (const char**) INFO_TYPES;
-    content->infosList.infoContents = (const char**) INFO_CONTENTS;
+    
+    // the first settings page contains only the version and the developer name
+    // of the app (shall be always on the first setting page) 
+    if (page == 0) {
+        content->type = INFOS_LIST;
+        content->infosList.nbInfos = 2;
+        content->infosList.infoTypes = (const char**) INFO_TYPES;
+        content->infosList.infoContents = (const char**) INFO_CONTENTS;
+    }
+    // the second settings page contains 2 toggle setting switches
+    else if (page == 1) {
+        switches[DUMMY_SWITCH_1_ID].initState = (nbgl_state_t)N_storage.dummy1Allowed;
+        switches[DUMMY_SWITCH_1_ID].text = "Dummy 1";
+        switches[DUMMY_SWITCH_1_ID].subText = "Allow dummy 1\nin transactions";
+        switches[DUMMY_SWITCH_1_ID].token = DUMMY_SWITCH_1_TOKEN;
+        switches[DUMMY_SWITCH_1_ID].tuneId = TUNE_TAP_CASUAL;
+
+        switches[DUMMY_SWITCH_2_ID].initState = (nbgl_state_t)N_storage.dummy2Allowed;
+        switches[DUMMY_SWITCH_2_ID].text = "Dummy 2";
+        switches[DUMMY_SWITCH_2_ID].subText = "Allow dummy 2\nin transactions";
+        switches[DUMMY_SWITCH_2_ID].token = DUMMY_SWITCH_2_TOKEN;
+        switches[DUMMY_SWITCH_2_ID].tuneId = TUNE_TAP_CASUAL;
+
+        content->type = SWITCHES_LIST;
+        content->switchesList.nbSwitches = SETTINGS_SWITCHES_NB;
+        content->switchesList.switches = (nbgl_layoutSwitch_t*)switches;
+    }
+    else {
+        return false;
+    }
+    // valid page so return true
     return true;
 }
 
-void ui_menu_about() {
-#define TOTAL_PAGE_NB        (1)
-#define INIT_PAGE_INDEX      (0)
+// callback for setting warning choice
+ static void review_warning_choice(bool confirm) {
+    uint8_t switch_value;
+    if (confirm) 
+    {
+        // toggle the switch value
+        switch_value = !N_storage.dummy2Allowed;
+        // store the new setting value in NVM
+        nvm_write((void *)&N_storage.dummy2Allowed, &switch_value, 1);
+    }
+
+    // return to the settings menu
+    ui_menu_settings();
```

Comment:
Should we return to the second page of the setting menu here?

Reply:
yes feasible. But I prefer to keep the example simple and to avoid to change dynamically the init page